app-drawer/app-drawer.html

```diff
@@ -550,6 +567,66 @@
         }
       },
 
+      _addKeyboardFocusTrap: function() {
+        if (this.noFocusTrap) {
+          return;
+        }
+
+        this._lastFocusedElement = document.activeElement;
```

Comment:
Under native Shadow DOM, `document.activeElement` returns the outermost element.  e.g. focus anything inside publishing app `document.activeElement` will always return `blog-app`.  Fortunately we can still get the information from shadow root (https://www.w3.org/TR/2015/WD-shadow-dom-20151215/#active-element).  We may need to do something similar like this: https://github.com/PolymerElements/iron-overlay-behavior/blob/master/iron-overlay-manager.html#L68